lib/cisco_node_utils/router_ospf_area.rb

```diff
@@ -0,0 +1,285 @@
+#
+# NXAPI implementation of Router OSPF Area class
+#
+# June 2016, Sai Chintalapudi
+#
+# Copyright (c) 2016 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require 'ipaddr'
+require_relative 'node_util'
+require_relative 'router_ospf'
+require_relative 'router_ospf_vrf'
+
+module Cisco
+  # node_utils class for ospf_area
+  class RouterOspfArea < NodeUtil
+    attr_reader :router, :vrf, :area_id
+
+    def initialize(ospf_router, vrf_name, area_id)
+      fail TypeError unless ospf_router.is_a?(String)
+      fail TypeError unless vrf_name.is_a?(String)
+      fail TypeError unless area_id.is_a?(String)
+      fail ArgumentError unless ospf_router.length > 0
+      fail ArgumentError unless vrf_name.length > 0
+      fail ArgumentError unless area_id.length > 0
+
+      # check the area_id is a proper ipv4 address
+      fail ArgumentError if (begin
+                               IPAddr.new(area_id)
+                             rescue
+                               nil
+                             end).nil?
+      fail ArgumentError unless IPAddr.new(area_id).ipv4?
```

Comment:
Area IDs can be a simple Integer so these IPAddr calls are wrong.

I would just to_s all of the checks above and get rid of the IPAddr checks, i.e.:

```
-      fail TypeError unless ospf_router.is_a?(String)
-      fail TypeError unless vrf_name.is_a?(String)
-      fail TypeError unless area_id.is_a?(String)
-      fail ArgumentError unless ospf_router.length > 0
-      fail ArgumentError unless vrf_name.length > 0
-      fail ArgumentError unless area_id.length > 0
-      # check the area_id is a proper ipv4 address
-      fail ArgumentError if (begin
-                               IPAddr.new(area_id)
-                             rescue
-                               nil
-                             end).nil?
-      fail ArgumentError unless IPAddr.new(area_id).ipv4?
-      @router = ospf_router
-      @vrf = vrf_name
-      @area_id = area_id
+      @router = ospf_router.to_s
+      @vrf = vrf_name.to_s
+      @area_id = area_id.to_s
+      fail ArgumentError if @router.empty? || @vrf.empty? || @area_id.empty?
```